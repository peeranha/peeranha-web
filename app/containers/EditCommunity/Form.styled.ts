import {
  BG_PRIMARY,
  BORDER_RADIUS_M,
  BUTTON_COLOR,
  TEXT_LIGHT,
  TEXT_PRIMARY,
} from 'style-constants';
import { singleCommunityColors, singleCommunityStyles } from 'utils/communityManagement';

const communityColors = singleCommunityColors();
const communityStyles = singleCommunityStyles();

const styles = {
  tabsBlock: {
    borderBottom: `1px solid ${communityColors.dividerColor || 'rgba(216, 216, 216, 1)'}`,
    paddingBottom: '11px',
    whiteSpace: 'nowrap',
  },

  mainBlock: {
    margin: '32px',
    padding: '0 0 18px 0',

    '@media (min-width: 577px)': {
      borderBottom: `1px solid ${communityColors.dividerColor || 'rgba(216, 216, 216, 1)'}`,
    },
  },

  translationsTitle: {
    lineHeight: '23px',
    color: 'rgba(40, 40, 40, 1)',
  },

  translationsText: {
    color: communityColors.secondaryTextColor || 'rgba(123, 123, 123, 1)',
    lineHeight: '20px',
  },

  button: {
    padding: '3px 16px 4px',
    margin: '0 4px',
    lineHeight: '25px',
    color: `${communityColors.linkColor || TEXT_PRIMARY}`,
    transition: '0.5s',
  },

  activeTab: {
    color: 'rgb(255,255,255)',
    borderRadius: `${communityStyles.buttonBorderRadius || BORDER_RADIUS_M}`,
    background: `${communityColors.linkColor || BG_PRIMARY}`,
  },

  saveButton: {
    display: 'none !important',

    '@media (min-width: 577px)': {
      display: 'block !important',
      margin: '0 0 32px 181px !important',
    },
  },

  popupMenu: {
    background: '#FFFFFF',
    boxShadow: '0px -2px 4px rgba(7, 16, 64, 0.1)',
    zIndex: 10,

    '@media (min-width: 577px)': {
      display: 'none',
    },
  },

  popupMenuCloseButton: {
    width: '43vw',
    height: '40px',
    border: `1px solid ${communityColors.btnHeaderColor || BUTTON_COLOR}`,
    borderRadius: BORDER_RADIUS_M,
    fontSize: '16px',
    lineHeight: '20px',
    color: communityColors.newPostButtonText || BUTTON_COLOR,
    padding: '9px 16px 11px',
    transition: '0.5s',

    ':hover': {
      background: communityColors.btnHeaderColor || BUTTON_COLOR,
      color: communityColors.newPostButtonText || TEXT_LIGHT,
      border: `1px solid ${communityColors.btnHeaderColor || BUTTON_COLOR}`,
    },
  },

  popupMenuSaveButton: {
    width: '43vw',
    height: '40px',
    transition: '0.2s',
    background: communityColors.btnHeaderColor || BUTTON_COLOR,
    color: communityColors.newPostButtonText || TEXT_LIGHT,

    ':hover': {
      opacity: 0.8,
    },
    borderRadius: BORDER_RADIUS_M,
    padding: '9px 16px 11px',
  },
};

export default styles;
