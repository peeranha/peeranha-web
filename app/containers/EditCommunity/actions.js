/*
 *
 * EditCommunity actions
 *
 */

import {
  EDIT_COMMUNITY,
  EDIT_COMMUNITY_ERROR,
  EDIT_COMMUNITY_SUCCESS,
  GET_COMMUNITY,
  GET_COMMUNITY_ERROR,
  GET_COMMUNITY_SUCCESS,
} from './constants';

export function editCommunity(communityId, communityData, wallet) {
  return {
    type: EDIT_COMMUNITY,
    communityId,
    communityData,
    wallet,
  };
}

export function editCommunitySuccess() {
  return {
    type: EDIT_COMMUNITY_SUCCESS,
  };
}

export function editCommunityError(error) {
  return {
    type: EDIT_COMMUNITY_ERROR,
    error,
  };
}

export function getCommunity(communityId) {
  return {
    type: GET_COMMUNITY,
    communityId,
  };
}

export function getCommunitySuccess(community) {
  return {
    type: GET_COMMUNITY_SUCCESS,
    community,
  };
}

export function getCommunityError(error) {
  return {
    type: GET_COMMUNITY_ERROR,
    error,
  };
}
