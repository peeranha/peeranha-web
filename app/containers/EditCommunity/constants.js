/*
 *
 * EditCommunity constants
 *
 */

export const EDIT_COMMUNITY_FORM = 'app/EditCommunity/EDIT_COMMUNITY_FORM';
export const EDIT_COMMUNITY_BUTTON = 'app/EditCommunity/EDIT_COMMUNITY_BUTTON';

export const COMM_AVATAR_FIELD = 'app/EditCommunity/COMM_AVATAR_FIELD';
export const COMM_NAME_FIELD = 'app/EditCommunity/COMM_NAME_FIELD';
export const COMM_SHORT_DESCRIPTION_FIELD = 'app/EditCommunity/COMM_SHORT_DESCRIPTION_FIELD';
export const COMM_OFFICIAL_SITE_FIELD = 'app/EditCommunity/COMM_OFFICIAL_SITE_URL';
export const COMM_PEERANHA_SITE_FIELD = `app/EditCommunity/COMM_PEERANHA_SITE_FIELD`;
export const COMM_TRANSLATIONS_FIELD = 'app/EditCommunity/COMM_TRANSLATIONS_FIELD';
export const COMM_TRANSLATIONS_TITLE_FIELD = 'app/EditCommunity/COMM_TRANSLATIONS_TITLE_FIELD';
export const COMM_TRANSLATIONS_DESCRIPTION_FIELD =
  'app/EditCommunity/COMM_TRANSLATIONS_DESCRIPTION_FIELD';
export const COMM_AUTOTRANSLATIONS_FIELD = 'app/EditCommunity/COMM_AUTOTRANSLATIONS_FIELD';

export const EDIT_COMMUNITY = 'app/EditCommunity/EDIT_COMMUNITY';
export const EDIT_COMMUNITY_ERROR = 'app/EditCommunity/EDIT_COMMUNITY_ERROR';
export const EDIT_COMMUNITY_SUCCESS = 'app/EditCommunity/EDIT_COMMUNITY_SUCCESS';

export const GET_COMMUNITY = 'app/EditCommunity/GET_COMMUNITY';
export const GET_COMMUNITY_ERROR = 'app/EditCommunity/GET_COMMUNITY_ERROR';
export const GET_COMMUNITY_SUCCESS = 'app/EditCommunity/GET_COMMUNITY_SUCCESS';

export const GENERAL_TAB = 'general';
export const TRANSLATIONS_TAB = 'translations';
export const CHINESE_LANG = 'Chinese';
