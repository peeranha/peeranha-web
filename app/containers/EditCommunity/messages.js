/*
 * EditCommunity Messages
 *
 * This contains all the text for the EditCommunity component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  avatar: {
    id: 'app.containers.CreateCommunity.avatar',
  },
  communityTitle: {
    id: 'app.containers.CreateCommunity.communityTitle',
  },
  communityTitleTip: {
    id: 'app.containers.EditCommunity.communityTitleTip',
  },
  description: {
    id: 'app.containers.EditCommunity.description',
  },
  editCommunity: {
    id: 'app.containers.EditCommunity.editCommunity',
  },
  header: {
    id: 'app.containers.EditCommunity.header',
  },
  website: {
    id: 'app.containers.CreateCommunity.website',
  },
  websiteTip: {
    id: 'app.containers.CreateCommunity.websiteTip',
  },
  pageTitle: {
    id: 'app.containers.EditCommunity.pageTitle',
  },
  shortDescription: {
    id: 'app.containers.CreateCommunity.shortDescription',
  },
  shortDescriptionTip: {
    id: 'app.containers.CreateCommunity.shortDescriptionTip',
  },
  whyWeNeedIt: {
    id: 'app.containers.CreateCommunity.whyWeNeedIt',
  },
  whyWeNeedItTip: {
    id: 'app.containers.CreateCommunity.whyWeNeedItTip',
  },
});
