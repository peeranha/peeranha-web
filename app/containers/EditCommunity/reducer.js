/*
 *
 * EditCommunity reducer
 *
 */

import { fromJS } from 'immutable';

import {
  EDIT_COMMUNITY,
  EDIT_COMMUNITY_ERROR,
  EDIT_COMMUNITY_SUCCESS,
  GET_COMMUNITY,
  GET_COMMUNITY_ERROR,
  GET_COMMUNITY_SUCCESS,
  FREEZE_COMMUNITY,
  FREEZE_COMMUNITY_ERROR,
  FREEZE_COMMUNITY_SUCCESS,
} from './constants';

export const initialState = fromJS({
  community: null,
  editCommunityError: null,
  editCommunityLoading: false,
  freezeCommunityError: null,
  freezeCommunityLoading: false,
  getCommunityError: null,
  getCommunityLoading: false,
});

export default (state = initialState, action) => {
  switch (action.type) {
    case GET_COMMUNITY:
      return state.set('getCommunityLoading', true);
    case GET_COMMUNITY_SUCCESS:
      return state.set('community', action.community).set('getCommunityLoading', false);
    case GET_COMMUNITY_ERROR:
      return state.set('getCommunityError', action.error).set('getCommunityLoading', false);
    case EDIT_COMMUNITY:
      return state.set('editCommunityLoading', true);
    case EDIT_COMMUNITY_SUCCESS:
      return state.set('editCommunityLoading', false);
    case EDIT_COMMUNITY_ERROR:
      return state.set('editCommunityError', action.error).set('editCommunityLoading', false);
    case FREEZE_COMMUNITY:
      return state.set('freezeCommunityLoading', true);
    case FREEZE_COMMUNITY_SUCCESS:
      return state.set('freezeCommunityLoading', false);
    case FREEZE_COMMUNITY_ERROR:
      return state.set('freezeCommunityError', action.error).set('freezeCommunityLoading', false);
    default:
      return state;
  }
};
