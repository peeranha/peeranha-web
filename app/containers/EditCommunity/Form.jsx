import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form/immutable';
import * as routes from 'routes-config';
import { Link } from 'react-router-dom';
import { isSingleCommunityWebsite } from 'utils/communityManagement';

import {
  imageValidation,
  required,
  strLength3x20,
  strLength15x250,
  strLength100Max,
  validateURL,
  withoutDoubleSpace,
} from 'components/FormFields/validate';

import FormBox from 'components/Form';
import LargeButton from 'components/Button/Contained/InfoLarge';
import { ExtendedBase } from 'components/Base/AvatarBase';
import AvatarField from 'components/FormFields/AvatarField';
import TextInputField from 'components/FormFields/TextInputField';
import TextareaField from 'components/FormFields/TextareaField';
import TranslationsField from 'components/FormFields/TranslationsField';
import ScrollContainer from 'components/common/ScrollContainer';

import { scrollToErrorField } from 'utils/animation';
import { isSuiBlockchain } from 'utils/constants';

import WarningFreezeCommunityModal from './WarningFreezeCommunityModal';
import {
  COMM_AUTOTRANSLATIONS_FIELD,
  COMM_AVATAR_FIELD,
  COMM_NAME_FIELD,
  COMM_OFFICIAL_SITE_FIELD,
  COMM_PEERANHA_SITE_FIELD,
  COMM_SHORT_DESCRIPTION_FIELD,
  COMM_TRANSLATIONS_DESCRIPTION_FIELD,
  COMM_TRANSLATIONS_FIELD,
  COMM_TRANSLATIONS_TITLE_FIELD,
  EDIT_COMMUNITY_BUTTON,
  FREEZE_COMMUNITY_BUTTON,
  EDIT_COMMUNITY_FORM,
  GENERAL_TAB,
  TRANSLATIONS_TAB,
  ADVANCED_TAB,
} from './constants';
import { COMMUNITY_TYPE } from '../CreateCommunity/constants';
import styles from './Form.styled';

const isSingleCommunityMode = isSingleCommunityWebsite();

const EditCommunityForm = ({
  community,
  communityId,
  communityLoading,
  editCommunityDispatch,
  freezeCommunityLoading,
  freezeCommunityDispatch,
  handleSubmit,
  tab,
  setTab,
}) => {
  const { t } = useTranslation();
  const nextRoute = isSingleCommunityMode ? routes.feed : routes.communities;
  const commId = isSuiBlockchain ? community?.id : communityId;
  const [selectedLanguages, setSelectedLanguages] = useState([]);
  const editCommunity = useCallback(
    (values) => {
      const communityData = {
        avatar: values.get(COMM_AVATAR_FIELD),
        name: values.get(COMM_NAME_FIELD),
        description: values.get(COMM_SHORT_DESCRIPTION_FIELD),
        website: values.get(COMM_OFFICIAL_SITE_FIELD),
        communitySite: values.get(COMM_PEERANHA_SITE_FIELD),
        isBlogger: false,
        translations: selectedLanguages.map((language) => ({
          communityId: commId,
          id: `${commId}-${language}`,
          description: values.get(`${COMM_TRANSLATIONS_DESCRIPTION_FIELD}-${commId}-${language}`),
          enableAutotranslation: values.get(`${COMM_AUTOTRANSLATIONS_FIELD}-${commId}-${language}`),
          language,
          name: values.get(`${COMM_TRANSLATIONS_TITLE_FIELD}-${commId}-${language}`),
        })),
      };

      editCommunityDispatch(communityId, communityData);
    },
    [commId, editCommunityDispatch, selectedLanguages],
  );

  const setGeneralTab = (event) => {
    event.preventDefault();
    setTab(GENERAL_TAB);
  };
  const setTranslationsTab = (event) => {
    event.preventDefault();
    setTab(TRANSLATIONS_TAB);
  };
  const setAdvancedTab = (event) => {
    event.preventDefault();
    setTab(ADVANCED_TAB);
  };

  return (
    <FormBox onSubmit={handleSubmit(editCommunity)}>
      <ExtendedBase
        css={{
          ...styles.mainBlock,
          ...(tab === ADVANCED_TAB && {
            '@media (min-width: 577px)': {
              borderBottom: 'none',
            },
          }),
        }}
      >
        <Field
          name={COMM_AVATAR_FIELD}
          component={AvatarField}
          validate={[imageValidation, required]}
          warn={[imageValidation, required]}
          disabled={communityLoading}
        />
        <div>
          <div className="mb24" css={styles.tabsBlock}>
            <ScrollContainer>
              <div className="mb12">
                <button
                  css={{
                    ...styles.button,
                    ...(tab === GENERAL_TAB && styles.activeTab),
                  }}
                  onClick={setGeneralTab}
                >
                  {t('common.editCommunityDesc.general')}
                </button>
                <button
                  css={{
                    ...styles.button,
                    ...(tab === TRANSLATIONS_TAB && styles.activeTab),
                  }}
                  onClick={setTranslationsTab}
                >
                  {t('common.editCommunityDesc.translations')}
                </button>
                <button
                  css={{
                    ...styles.button,
                    ...(tab === ADVANCED_TAB && styles.activeTab),
                  }}
                  onClick={setAdvancedTab}
                >
                  {t('common.editCommunityDesc.advanced')}
                </button>
              </div>
            </ScrollContainer>
          </div>
          <div>
            <div className={tab !== GENERAL_TAB ? 'dn' : ''}>
              <Field
                name={COMM_NAME_FIELD}
                component={TextInputField}
                validate={[withoutDoubleSpace, strLength3x20, required]}
                warn={[strLength3x20, required]}
                disabled={communityLoading}
                label={t('createCommunity.communityTitle')}
                splitInHalf
                tip={t('createCommunity.communityTitleTip')}
              />

              <Field
                name={COMM_SHORT_DESCRIPTION_FIELD}
                component={TextareaField}
                validate={[withoutDoubleSpace, strLength15x250, required]}
                warn={[strLength15x250, required]}
                disabled={communityLoading}
                label={t('createCommunity.shortDescription')}
                splitInHalf
                tip={t('createCommunity.shortDescriptionTip')}
              />

              <Field
                name={COMM_OFFICIAL_SITE_FIELD}
                component={TextInputField}
                validate={[withoutDoubleSpace, validateURL, strLength100Max]}
                warn={[validateURL]}
                disabled={communityLoading}
                label={t('createCommunity.website')}
                placeholder="https://example.com"
                splitInHalf
                tip={t('createCommunity.websiteTip')}
              />

              <Field
                disabled={communityLoading}
                name={COMM_PEERANHA_SITE_FIELD}
                component={TextInputField}
                label={t('createCommunity.communityWebsite')}
                validate={[withoutDoubleSpace, strLength100Max]}
                warn={[strLength100Max]}
                placeholder="subdomain.peeranha.io"
                tip={t('createCommunity.communityWebsiteTip')}
                splitInHalf
              />
            </div>
            <div className={tab !== TRANSLATIONS_TAB ? 'dn' : ''}>
              <span className="fz16 semi-bold pb8" css={styles.translationsTitle}>
                {t('common.editCommunityDesc.interfaceLanguages')}
              </span>
              <p className="fz16 mb12" css={styles.translationsText}>
                {t('common.editCommunityDesc.interfaceTranslation')}
              </p>

              <Field
                disabled={communityLoading}
                name={COMM_TRANSLATIONS_FIELD}
                component={TranslationsField}
                translations={community?.translations}
                communityId={communityId}
                selectedLanguages={selectedLanguages}
                setSelectedLanguages={setSelectedLanguages}
              />
            </div>
            <div className={isSuiBlockchain || tab !== ADVANCED_TAB ? 'dn' : ''}>
              <span className="fz16 semi-bold pb8" css={styles.translationsTitle}>
                {t('common.editCommunityDesc.emergencyFreeze')}
              </span>
              <WarningFreezeCommunityModal
                isFrozen={community?.isFrozen}
                submitAction={() => freezeCommunityDispatch(community?.isFrozen, communityId)}
                Button={({ onClick }) => (
                  <LargeButton
                    id={FREEZE_COMMUNITY_BUTTON}
                    onClick={(e) => {
                      e.preventDefault();
                      onClick();
                    }}
                    disabled={freezeCommunityLoading}
                    block={freezeCommunityLoading}
                    css={styles.freezeButton}
                  >
                    {t(
                      `common.editCommunityDesc.${
                        community?.isFrozen ? 'unfreezeCommunity' : 'freezeCommunity'
                      }`,
                    )}
                  </LargeButton>
                )}
              />
            </div>
          </div>
        </div>
      </ExtendedBase>
      {tab !== ADVANCED_TAB && (
        <LargeButton
          id={EDIT_COMMUNITY_BUTTON}
          type="submit"
          disabled={communityLoading}
          css={styles.saveButton}
        >
          {t('common.editCommunityDesc.editCommunity')}
        </LargeButton>
      )}
      <div
        className={`${tab === ADVANCED_TAB ? 'dn' : 'pf b0 l0 full-width'}`}
        css={styles.popupMenu}
      >
        <div className="df jcsb pr16 pl16 pt8 pb8">
          <Link to={nextRoute()}>
            <button css={styles.popupMenuCloseButton}>{t('common.close')}</button>
          </Link>
          <button css={styles.popupMenuSaveButton} type="submit" disabled={communityLoading}>
            {t('common.editCommunityDesc.editCommunity')}
          </button>
        </div>
      </div>
    </FormBox>
  );
};

EditCommunityForm.propTypes = {
  communityId: PropTypes.number.isRequired,
  communityLoading: PropTypes.bool.isRequired,
  editCommunityDispatch: PropTypes.func.isRequired,
  freezeCommunityLoading: PropTypes.bool.isRequired,
  freezeCommunityDispatch: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

const FormClone = reduxForm({
  enableReinitialize: true,
  form: EDIT_COMMUNITY_FORM,
  onSubmitFail: (errors) => {
    scrollToErrorField(errors);
  },
})(EditCommunityForm);

export default connect((state, { community }) => {
  const translationsFields = {};
  if (community) {
    for (let i = 0; i < community.translations.length; i++) {
      translationsFields[`${COMM_TRANSLATIONS_TITLE_FIELD}-${community.translations[i].id}`] =
        community.translations[i].name;
      translationsFields[`${COMM_TRANSLATIONS_DESCRIPTION_FIELD}-${community.translations[i].id}`] =
        community.translations[i].description;
      translationsFields[`${COMM_AUTOTRANSLATIONS_FIELD}-${community.translations[i].id}`] =
        community.translations[i].enableAutotranslation;
    }
  }

  return {
    formValues: state.toJS()?.form[EDIT_COMMUNITY_FORM]?.values ?? {},
    initialValues: community
      ? {
          [COMM_AVATAR_FIELD]: community.avatar,
          [COMM_AVATAR_FIELD]: community.avatar,
          [COMM_NAME_FIELD]: community.name,
          [COMM_SHORT_DESCRIPTION_FIELD]: community.description,
          [COMM_OFFICIAL_SITE_FIELD]: community.website,
          [COMM_PEERANHA_SITE_FIELD]: community.communitySite,
          [COMMUNITY_TYPE]: 0,
          ...translationsFields,
        }
      : {},
  };
})(FormClone);
