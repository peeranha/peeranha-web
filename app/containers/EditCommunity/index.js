/**
 *
 * EditCommunity
 *
 */

import React, { useEffect, useMemo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { translationMessages } from 'i18n';
import { createStructuredSelector } from 'reselect';
import { compose, bindActionCreators } from 'redux';

import LoadingIndicator from 'components/LoadingIndicator/WidthCentered';
import Seo from 'components/Seo';
import TipsBase from 'components/Base/TipsBase';

import {
  WHAT_IS_COMMUNITY_QUESTION,
  WHO_MANAGES_COMMUNITY_QUESTION,
} from 'containers/Faq/constants';

import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
import { makeSelectProfileInfo } from 'containers/AccountProvider/selectors';
import { selectFaqQuestions } from 'containers/DataCacheProvider/selectors';

import Header from 'containers/CreateCommunity/Header';
import Tips from 'containers/CreateCommunity/Tips';

import { noAccess as noAccessRoute } from 'routes-config';

import { communityModeratorCreatePermission } from 'utils/properties';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import Form from './Form';

import { editCommunity, getCommunity } from './actions';
import {
  selectCommunity,
  selectEditCommunityLoading,
  selectGetCommunityLoading,
} from './selectors';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';

const EditCommunity = ({
  community,
  editCommunityDispatch,
  editCommunityLoading,
  faqQuestions,
  getCommunityDispatch,
  locale,
  profileInfo,
  getCommunityLoading: communityLoading,
  match: {
    params: { communityId },
  },
}) => {
  const editingAllowed = useMemo(
    () =>
      communityModeratorCreatePermission(
        profileInfo?.['integer_properties'] || [],
      ),
    [profileInfo],
  );

  if (!editingAllowed) {
    return <Redirect to={noAccessRoute()} />;
  }

  useEffect(
    () => {
      getCommunityDispatch(communityId);
    },
    [communityId],
  );

  const formData = useMemo(
    () => ({
      community,
      editCommunityDispatch,
      communityId: +communityId,
      communityLoading: editCommunityLoading,
      locale,
    }),
    [community, communityId, editCommunityDispatch, editCommunityLoading],
  );

  const translations = useMemo(() => translationMessages[locale], [locale]);

  return (
    <div>
      <Seo
        title={translations[messages.pageTitle.id]}
        description={translations[messages.description.id]}
        index={false}
        language={locale}
      />

      <Header headerDescriptor={messages.header} />

      <TipsBase className="overflow-hidden">
        {communityLoading && <LoadingIndicator />}

        {!communityLoading && <Form {...formData} />}

        <Tips faqQuestions={faqQuestions} />
      </TipsBase>
    </div>
  );
};

EditCommunity.propTypes = {
  editCommunityDispatch: PropTypes.func.isRequired,
  editCommunityLoading: PropTypes.bool.isRequired,
  getCommunityDispatch: PropTypes.func.isRequired,
  getCommunityLoading: PropTypes.bool.isRequired,
  locale: PropTypes.string.isRequired,
  match: PropTypes.object.isRequired,
  profileInfo: PropTypes.object.isRequired,
  community: PropTypes.object,
  faqQuestions: PropTypes.array,
};

const withConnect = connect(
  createStructuredSelector({
    community: selectCommunity(),
    editCommunityLoading: selectEditCommunityLoading(),
    faqQuestions: selectFaqQuestions([
      WHAT_IS_COMMUNITY_QUESTION,
      WHO_MANAGES_COMMUNITY_QUESTION,
    ]),
    getCommunityLoading: selectGetCommunityLoading(),
    locale: makeSelectLocale(),
    profileInfo: makeSelectProfileInfo(),
  }),
  dispatch => ({
    editCommunityDispatch: bindActionCreators(editCommunity, dispatch),
    getCommunityDispatch: bindActionCreators(getCommunity, dispatch),
  }),
);

export default compose(
  injectReducer({ key: 'editcommunity', reducer }),
  injectSaga({ key: 'editcommunity', saga }),
  withConnect,
)(EditCommunity);
