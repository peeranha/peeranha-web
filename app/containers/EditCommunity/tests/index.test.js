import React from 'react';
import { Provider } from 'react-redux';
import { memoryHistory } from 'react-router-dom';
import { ConnectedRouter } from 'react-router-redux';
import { render } from '@testing-library/react';
import { translationMessages } from 'i18n';

import configureStore from 'configureStore';
import createdHistory from 'createdHistory';

import LanguageProvider from 'containers/LanguageProvider';

import EditCommunity from '../index';

jest.mock('react-avatar-edit', () => jest.fn(() => <mock-avatar />));

jest.mock('communities-config', () =>
  jest.fn(() => ({
    key: { origin: '' },
  })),
);

jest.mock('utils/properties', () => ({
  communityModeratorCreatePermission: jest.fn(() => true),
}));

jest.mock('containers/AccountProvider/selectors', () => ({
  makeSelectProfileInfo: jest.fn(() => () => ({
    integer_properties: [],
  })),
}));
jest.mock('containers/DataCacheProvider/selectors', () => ({
  selectCommunities: jest.fn(() => () => []),
  selectFaqQuestions: jest.fn(() => () => []),
}));
jest.mock('containers/LanguageProvider/selectors', () => ({
  makeSelectLocale: jest.fn(() => () => 'en'),
}));

jest.mock('../selectors', () => ({
  selectCommunity: jest.fn(() => () => ({
    avatar: 'Avatar',
    name: 'Name',
    description: 'Short description',
    website: '',
    main_description: 'Descriptive description',
  })),
  selectEditCommunityLoading: jest.fn(() => () => false),
  selectGetCommunityLoading: jest.fn(() => () => false),
}));

const props = {
  match: {
    params: { communityId: 1 },
  },
};

describe('<EditCommunity />', () => {
  const store = configureStore({}, memoryHistory);

  it('renders and matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <LanguageProvider locale="en" key="en" messages={translationMessages}>
          <ConnectedRouter history={createdHistory}>
            <EditCommunity match={props.match} />
          </ConnectedRouter>
        </LanguageProvider>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
