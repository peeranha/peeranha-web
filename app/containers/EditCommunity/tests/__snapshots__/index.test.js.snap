// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditCommunity /> renders and matches the snapshot 1`] = `
<div>
  <div
    class="mb-to-sm-0 mb-from-sm-3 Base-gxTqDr hgJIyR"
  >
    <h3
      class="f1znse-0-Span-juOiwt SaUZL"
    >
      <img
        alt="Community icon"
        class="Img-funbTP bVljxa"
        src="IMAGE_MOCK"
      />
      <span>
        Edit community
      </span>
    </h3>
    <div
      class="right-panel Simple__WrapperRightPanel-sc-1uv1n1f-0 kkNECD"
    >
      <a
        class="A-sc-1lkwpuz-1 bMWbep"
        href="/communities"
      >
        <button>
          <img
            alt="x"
            class="mr-1"
            src="IMAGE_MOCK"
          />
          <span
            class="button-label Span-f1znse-0 cOvmGD"
            color="#576fed"
          >
            <span>
              Close
            </span>
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    class="overflow-hidden Base-gxTqDr btHItu"
  >
    <div
      class="sc-1j4f184-0-Base-gxTqDr IMsVu"
    >
      <div
        class="AvatarField__Div-sc-1bmtuz8-0 fACQrM"
        id="app-EditCommunity-COMM-AVATAR-FIELD"
        value="6"
      >
        <div>
          <div>
            <div
              class="avatar-wrapper"
            >
              <mock-avatar />
              <button
                class="close-icon"
              />
            </div>
            <div
              class="reload-bg"
            />
          </div>
          <img
            alt="icon"
            class="sc-1swdnke-0-Img-funbTP hYsIYr"
            src="undefined://undefined/Avatar"
          />
        </div>
        <div
          class="AvatarField__InfoMessage-sc-1bmtuz8-1 bKZxys"
        >
          <span>
            File size should not exceed 5Mb
          </span>
        </div>
      </div>
      <form
        class="Form-sc-1vkcewz-0 dbpQQy"
      >
        <div
          class="Wrapper__StyledBox-rr64xs-0 gIQOlG"
          id="app-EditCommunity-COMM-NAME-FIELD"
        >
          <h6
            class="f1znse-0-Span-juOiwt bKHwNH"
          >
            Title
          </h6>
          <div>
            <div>
              <div
                class="InputStyled-sc-1xf9t49-0 fYDrLB"
              >
                <input
                  name="app/EditCommunity/COMM_NAME_FIELD"
                  type="text"
                  value="Name"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Wrapper__StyledBox-rr64xs-0 gIQOlG"
          id="app-EditCommunity-COMM-SHORT-DESCRIPTION-FIELD"
        >
          <h6
            class="f1znse-0-Span-juOiwt bKHwNH"
          >
            Short description
          </h6>
          <div>
            <div>
              <div
                class="InputStyled-sc-1xf9t49-0 fYDrLB"
              >
                <input
                  name="app/EditCommunity/COMM_SHORT_DESCRIPTION_FIELD"
                  type="text"
                  value="Short description"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Wrapper__StyledBox-rr64xs-0 gIQOlG"
          id="app-EditCommunity-COMM-OFFICIAL-SITE-URL"
        >
          <h6
            class="f1znse-0-Span-juOiwt bKHwNH"
          >
            Official site
          </h6>
          <div>
            <div>
              <div
                class="InputStyled-sc-1xf9t49-0 fYDrLB"
              >
                <input
                  name="app/EditCommunity/COMM_OFFICIAL_SITE_URL"
                  placeholder="https://example.com"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <button
          class="Button-kDSBcD knCmMg"
          id="app/EditCommunity/EDIT_COMMUNITY_BUTTON"
          type="submit"
        >
          Edit community
        </button>
      </form>
    </div>
    <div>
      <h6
        class="mb-3 f1znse-0-Span-juOiwt bKHwNH"
      >
        <span>
          Tips
        </span>
      </h6>
      <ul
        class="Tips__Ul-sc-1l950jl-1 jIOofx"
      >
        <li>
          <span>
            Image will be the face of community
          </span>
        </li>
        <li>
          <span>
            Specify a memorable title
          </span>
        </li>
        <li>
          <span>
            Community description should be concise and have a clear description for its users
          </span>
        </li>
        <li>
          <span>
            Describe how this community will attract the attention of other users and collect votes
          </span>
        </li>
      </ul>
      <ul />
    </div>
  </div>
</div>
`;
